entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/SiteEntityProvider.java

```diff
@@ -309,7 +310,11 @@ public ActionReturn handleGroups(EntityView view, Map<String, Object> params) th
         // check if the user can access site
         isAllowedAccessSite(site);
 
-        List<EntityGroup> groups = site.getGroups().stream().map(EntityGroup::new).collect(Collectors.toList());
+        List<EntityGroup> groups = securityService.isSuperUser() ?
+                site.getGroups().stream().map(EntityGroup::new).collect(Collectors.toList())
+                :
+                site.getGroupsWithMember(developerHelperService.getCurrentUserId()).stream()
```

Comment:
developerHelperService is deprecated and we try not to use it any more. Could you please use sessionManager.getCurrentSessionUserId() ? We should probably mark up the docs better for DeveloperHelperService.

Reply:
Done!